lib/sigstore/signer.rb

```diff
@@ -0,0 +1,246 @@
+# frozen_string_literal: true
+
+# Copyright 2024 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative "internal/util"
+require_relative "internal/x509"
+require_relative "models"
+require_relative "policy"
+require_relative "verifier"
+
+module Sigstore
+  class Signer
+    include Loggable
+
+    def initialize(jwt:, trusted_root:)
+      @jwt = jwt
+      @trusted_root = trusted_root
+    end
+
+    def sign(payload)
+      # 2) generate a keypair
+      keypair = generate_keypair
+      # 3) generate a CreateSigningCertificateRequest
+      csr = generate_csr(keypair)
+      # 4) get a cert chain from fulcio
+      chain = fetch_cert(csr)
```

Comment:
only get the last cert? or exclude the root? see comments around https://docs.google.com/document/d/1kbhK2qyPPk8SLavHzYSDM8-Ueul9_oxIMVFuWMWKz0E/edit?disco=AAAAvzifgZo

Reply:
@woodruffw thoughts here?